### PR TITLE
Bikerack filtering

### PR DESCRIPTION
**Pull request by sourabh-nutonomy**
This PR implements the logic for bike-rack filtering by using the `points_in_box()` method.

### CODE REVIEW

**Comment by sourabh-nutonomy** [comment]
The added unit tests pass. I have checked it locally. I don't know why Jenkins is not automatically running the unit tests on the server.

File: python-sdk/nuscenes/utils/geometry_utils.py [hunk 1]
```diff
@@ -106,3 +106,40 @@ def transform_matrix(translation: np.ndarray = np.array([0, 0, 0]),
...
+    """
+    Checks whether points are inside the box.
+
+    Picks one corner as reference (p1) and computes the vector to a target point (v).
+    Then for each of the 3 axes, project v onto the axis and compare the length.
+    Inspired by: https://math.stackexchange.com/a/1552579
+    :param box: <Box>.
+    :param points: <np.float: 3, n>.
+    :param wlh_factor: Inflates or deflates the box.
+    :return: <np.bool: n, >.
```
Comment (holger-nutonomy) [inline python-sdk/nuscenes/utils/geometry_utils.py]: Add docstrings for parameters, adjust spacing around `float = 1.0`.

File: python-sdk/nuscenes/eval/detection/loaders.py [hunk 2]
```diff
@@ -117,6 +127,22 @@ def filter_eval_boxes(nusc, eval_boxes: EvalBoxes, max_dist: dict):
...
+        sample_anns = nusc.get('sample', sample_token)['anns']
+        bikerack_recs = [nusc.get('sample_annotation', ann) for ann in sample_anns if
+                         nusc.get('sample_annotation', ann)['category_name'] == 'static_object.bicycle_rack']
+
+        filtered_boxes = []
+
+        for rec in bikerack_recs:
+            bikerack_box = Box(rec['translation'], rec['size'], Quaternion(rec['rotation']))
+            for box in eval_boxes[sample_token]:
+                if box.detection_name in ['bicycle', 'motorcycle'] and \
```
Comment (holger-nutonomy) [inline python-sdk/nuscenes/eval/detection/loaders.py]: Include motorcycle?

**Comment by holger-nutonomy** [comment]
> The added unit tests pass. I have checked it locally. I don't know why Jenkins is not automatically running the unit tests on the server.

@chris-li-nutonomy

File: python-sdk/nuscenes/eval/detection/loaders.py (hunk 2 above)
Comment (Alex-nutonomy) [reply python-sdk/nuscenes/eval/detection/loaders.py]: According to the instructions https://github.com/nutonomy/nuscenes-devkit/blob/master/instructions.md#bicycle-rack, a bike rack only mentions bicycles. But I know from our review that sometimes motorcycles are parked inside bike racks.

File: python-sdk/nuscenes/eval/detection/tests/test_loader.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+        nusc = NuScenes(version='v1.0-mini', dataroot=os.environ['NUSCENES'], verbose=False)
+
+        sample_token = '0af0feb5b1394b928dd13d648de898f5'
+        # This sample has a bike rack instance 'bfe685042aa34ab7b2b2f24ee0f1645f' with these parameters
+        # 'translation': [683.681, 1592.002, 0.809],
+        # 'size': [1.641, 14.465, 1.4],
+        # 'rotation': [0.3473693995546558, 0.0, 0.0, 0.9377283723195315]
+
+        max_dist = {'car': 50,
```
Comment (oscar-nutonomy) [inline python-sdk/nuscenes/eval/detection/tests/test_loader.py]: You could get this from the config factory in `config.py` also if you like.

Comment (sourabh-nutonomy) [reply python-sdk/nuscenes/eval/detection/tests/test_loader.py]: Done. Thanks!

**Pull request merged**